handlers.go

```diff
@@ -225,3 +220,62 @@ func attestationHandler(useProfiling bool, hashes *AttestationHashes) http.Handl
 		fmt.Fprintln(w, b64Doc)
 	}
 }
+
+// leaderHandler is called when the enclave is designated as leader enclave.
+// If designated, we do the following:
+//
+//  1. Signal to our leader registration goroutine that we're the leader.
+//  2. Start the worker event loop, to keep track of worker enclaves.
+//  3. Expose leader-specific endpoints.
+func leaderHandler(ctx context.Context, e *Enclave) http.HandlerFunc {
+	var once sync.Once
+	return func(w http.ResponseWriter, r *http.Request) {
+		once.Do(func() {
+			e.becameLeader <- struct{}{}
+			go e.workers.start(ctx)
+			// Make leader-specific endpoints available.
+			e.intSrv.Handler.(*chi.Mux).Put(pathState, putStateHandler(e))
+			e.extPrivSrv.Handler.(*chi.Mux).Post(pathHeartbeat, heartbeatHandler(e))
+			elog.Println("Designated enclave as leader.")
+		})
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func heartbeatHandler(e *Enclave) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var (
+			hb              heartbeatRequest
+			syncAndRegister = func(keys *enclaveKeys, worker *url.URL) {
+				if err := asLeader(keys.get()).syncWith(worker); err == nil {
```

Comment:
verified that this call re-initiates the full synchronization protocol and verifies attestation documents again.